### DISCUSSION TITLE
RGB control with tasmota

### DISCUSSION THREAD

**Discussion by lionlggnd** [arendst/Tasmota#17834] [Q&A] [answered]
Hi, I would like to use an ESP8266 nodemcu devkit to control a 5V rgb led strip.
on the internet there are various and valid softwares to do that, but i decided to use tasmota because i have other devices (sonoff,esp etc.) running on Tasmota.

I want to know how I could configure GPIO and tasmota template to do this.
I don't think that the 5V output of the nodemcu is a PWM output, so I'm also satisfied that the LED always remains at its maximum brightness, as long as it knows how to change color and can be controlled via web interface. Mentally I thought that this could be achieved by connecting the positive of the led strip to the 5v output of the nodemcu and the negative (R,G,B) to the GPIOs.
Let me know how i can configure that and some tip that I may have missed.

**Comment by sfromis** [comment]
Using NodeMCU, you cannot drive a led strip without additional components. Such a strip would generally have one connection directly to a 5V power source, with "ground" for each of the channels going through a driver circuit, typically having a MOSFET to modulate the power based on a weak PWM signal from the MCU. When you have the hardware done right, the Tasmota config for the GPIOs would simply be PWM channels. Note that if the strip has addressable leds (RGBIC), it is an entirely different situation.

The hardware build for this is really simple (many examples on the net), but you can of course also get commercial devices with both MCU and driver circuits to run Tasmota on.

NoceMCU has no 5V output, but has a pin which may be connected to the 5V input, but indeed not something which the ESP8266 has any access to.

**Comment by lionlggnd** [comment]
some cards, if I'm not mistaken, have connections that carry 5v from the usb input.

I wanted to connect the positive of the LED strip to this pin, even if it doesn't exist, I will get the 5V from myself.
As for the ground pole, isn't the simple GPIO PWM connection good?
If not, you tell me the circuit that I have to do.
Thank you very much


my rgb strip is simple rgb, not RGBW or addressable.

**Reply by sfromis** [reply]
Yes, as I already wrote:
> but has a pin which may be connected to the 5V input

My point was that it not an output, which would be the NodeMCU generating 5V in case of input voltage being higher, just a connection. As long as you do not draw more than 1A, it should be ok to take your USB input voltage from the NodeMCU pin.

> As for the ground pole, isn't the simple GPIO PWM connection good?

As I specifically explained to you already, no. You need a MOSFET to manage the current, the ESP8266 cannot handle current. If you Google for something like `esp8266 mosfet strip`, you'll find examples of simple builds.

**Comment by lionlggnd** [comment]
![image](https://user-images.githubusercontent.com/117315249/215595071-8ba10e7f-216c-42cc-a7e6-3ca09727432f.png)

this can go?
is more easier if i build a simple pcb with a online pcb printing service that do this.
as I see, it is a question of connecting the first pole of each mosfet to the various outputs (r, g, b) of the nodemcu, the central pole of the mosfet will bring the negative rgb outputs modulated to the led strip. the third and last pin is simple GND and no use
right?

**Reply by barbudor** [reply]
Assuming the standard pinout of a FET N as below, this looks like pretty much it

![image](https://user-images.githubusercontent.com/2996042/215603686-3f80216e-3f75-4ee6-a29a-777020611aa6.png)

**Comment by lionlggnd** [comment]
after that, how i can configure the template in tsmota?

**Reply by barbudor** [reply]
In Configure -> Configure Module, set the GPIO for D2, D3, D4 to PWM wiht respective index 1, 2, 3

![image](https://user-images.githubusercontent.com/2996042/215604236-96674a43-9dce-4618-bcd6-8aefc77c391d.png)

Note that D3 (GPIO0) and D4 (GPIO2) have pull-ups that are mandatory for proper boot of the ESP
So during boot or restart, the GPIOs HI which will create a Yellow (Red+Green) flash which may be not very nice

I would suggest to move to ohter pins and add an external pull down (1K) between the GPIO and GND to insure that the GATE will not be driven high during boots

You could just mimic the pinout of existing products like MagicHome controllers : https://templates.blakadder.com/magichome_ZJ-WFMN-A_V1_1.html
Using GPIO5, 12 and 13
If it works for them, it will work for you

**Reply by sfromis** [reply]
As a starting point, I'd use gpios from 4,5,12,13,14. While others are also possible to use, they do have special considerations you may need to worry about.

**Reply by lionlggnd** [reply]
thanks. tasmota will auto recognize it as a rgb ?

**Reply by sfromis** [reply]
Yup

**Comment by lionlggnd** [comment]
excuse my ignorance, but what should I do with the last pin of the mosfet? I didn't quite understand. do i have to bring it to mass?

**Reply by sfromis** [reply]
You should follow the drawing you found. One pin ground, one to the gpio, and one to the color channel of the strip.

**Reply by lionlggnd** [reply]
ok, i'll bring it to mass.

**Comment by lionlggnd** [comment]
![image](https://user-images.githubusercontent.com/117315249/215608270-18b73fd5-211c-43bd-9587-6fcafae82130.png)
So this PCB i created now can work?

**Reply by barbudor** [reply]
You haven't changed the GPIO as you were advised to
Once you have changed the GPIO
add a pull-down resistor between the GPIO and GND

![image](https://user-images.githubusercontent.com/2996042/215609414-6e1f1fc5-ee34-4e92-b6bd-f077702594f3.png)

**Reply by lionlggnd** [reply]
Yes, thanks, i'll change it now. 😊
what gpios you suggest me?

**Reply by lionlggnd** [reply]
is it mandatory to add the resistance? because I have to find it

**Reply by barbudor** [reply]
Gpios have already been suggested by sfromis and myself

Resistors are better to avoid flashes. Highly recommended. Value is not critical. Almost any value between 1K and 47K would work

**Reply by sfromis** [reply]
Looks workable, as long as you make sure to to have 5V to the strip too. And indeed the resistors are a good addition; it will not break without, just possibly a bit of odd behavior.

You did not follow my recommendation of primarily using gpio 4,5,12,13,14 (which is also D2,D1,D6,D7,D8). Avoiding the encumbered pins does give more flexibility, like if you want to use the onboard led on gpio2 for something else, or if you want to use the button on gpio0 while also having RGB undisturbed. I prefer to only use gpio 0-3 and 15-16 if the easier ones are already taken.

I'd test the setup on the bench before getting a PCB made.

In case of your strip using more than 1A of power, I'd prefer having both the NodeMCU and the strip have parallel connections to power and ground from the power source, instead of routing the strip power trough the NodeMCU board traces. If less than 1A, you can do without this separation. If you go for a 12V strip instead of a 5V strip, you should of course have power separated, but still common ground.

**Comment by Noschvie** [comment]
Maybe this helps : https://github.com/zoesn89/WLED_PCB_D1_Mini

**Reply by lionlggnd** [reply]
Yes but this pcb is for ARGB strip. I have rgb

**Comment by lionlggnd** [comment]
![image](https://user-images.githubusercontent.com/117315249/215728571-d6affb07-1537-47e6-9036-7f94d1867f2e.png)
now it should work?

**Reply by barbudor** [reply]
Pull down on each gpio/gate conection

**Reply by barbudor** [reply]
I don't see how you will have a common ground with the leds and the power supply. You probably need more connectors and thicker tracks

**Reply by lionlggnd** [reply]
my rgb led is less than 1a so i pick + and - from the 5v usb connector.
my rgb strip is 5v.

**Comment by lionlggnd** [comment]
![image](https://user-images.githubusercontent.com/117315249/215748201-da1e8247-37a9-435a-9d07-b4f55e56d7a8.png)
should it go now? the tracks in blue are the ones on the bottom layer, since I didn't have space on the top layer to drive all the gpio outputs to ground with 1kohm.

**Reply by barbudor** [reply]
yes

**Reply by lionlggnd** [reply]
Thanks!! 😁😁

**Comment by joba-1** [comment]
Hi,

since you seem very new in this domain (nothing wrong with that), let me try to avoid some pitfalls I still see. 
You don't try something exotic, so what you have is likely to work, but not 100%.

You did not mention which mosfet you will use. There are a _lot_ of them available with different specifications, and you should definitely check their datasheet before ordering/using them (or post the exact type of your candidate here). Look for 
* Pin assignment. 1=Gate, Drain, Source=3 is no standard to really rely on.
* Switching mosfets with 3.3V as provided by the ESP is not very common: look at R_ds value or diagram in the datasheet. The value itself is not so relevant for your case, but the voltage V_ds given that is needed to get this R_ds. It is very often above the 3.3V but should at least be close. If your R_ds value is 4.5-6V you cannot reach full brightness (but might still be ok). Above that mosfets might not switch on at all at 3.3V.
* You have the resistors that avoid the leds flashing while the pins of the ESP have no defined voltage during early boot (or you don't want to care). But it might be important to protect the 3 ESP pwm pins as well. Insert resistors between Dx and the junction to your 1k resistors. Why? The gate of the mosfet acts as a small capacitor (again, some mosfets more than others). If you switch the voltage, the capacitor can deliver its charge back to the pins at a high current (although very short), well above what the esp pins are designed for. With ohms law you can check: you want at most ~20mA or so and you have 5V -> R=U/I = 5/0.02 = 250 ohm. The ESP might tolerate it without resistors, but why take chances. I'd go for the common values 220 or 330 ohm.
* Normal usb ports of pc's are specified to deliver up to 0.5A. You say your strip needs 1A, so take care where you plug it in. Check that your power supply can deliver that. Probably all non pc usb port power sources can do more, but check it anyways.

Have fun!

**Reply by lionlggnd** [reply]
thanks
i wanted to use irf510 as specified by some online websites.
you suggest me to change the 1k ohm resistors value to 220?
yes, i'll already have a power supply that can provide 1A+ power that i wanted to use with the ESP
Thanks for your suggestions!

**Reply by joba-1** [reply]
you’re welcome. 
No, I mean in addition to the 1k resistors.
But I‘d change the existing resistors from 1k to ~3-5k to make the difference to the new ones bigger (they build a voltage divider EspPinAt3.3V-Rnew-Gate-R1k-Gnd with the new ones and the higher resistance reduces the 3.3V at the gate a lot less)

**Reply by joba-1** [reply]
look here: https://www.vishay.com/docs/91015/irf510.pdf

Product summary says V_ds needs to be 10V for low resistance (i.e. mosfet is fully „on“) 
V_gs below 4V is not even shown in the I_d/V_gs-diagrams on page 3. 
-> With only 3.3V it will still be closed.

Not good. 
Unfortunately I don't know a better one right now.

**Reply by lionlggnd** [reply]
so i can use this mosfet?

**Reply by joba-1** [reply]
This one has the required specs but you might not like the form factor?

https://www.vishay.com/docs/66589/si7858bd.pdf

as i said to your selection: not good. Very likely the leds will stay completely dark

**Reply by lionlggnd** [reply]
omg wtf haha how can i connect that to my pcb
but if its the only option, a system i can find...
This MOSFETs can work with my configuration?
do you have any alternatives?

**Reply by barbudor** [reply]
A common N-channel that works well is NDP6020
Cheap on Ali, you can get 10pcs for around 3€ shipping included

Low VGSth < 1V
Low RDSon < 0.032 Ohm

![image](https://user-images.githubusercontent.com/2996042/215867329-aeaaff22-4c15-4e04-84f5-ab71ae10ccf1.png)

**Reply by lionlggnd** [reply]
so i can use this in my PCB?

**Reply by barbudor** [reply]
Yes, take care of purchaing the TO220 package. It is compatible with your PCB routing
the other is SMD

![image](https://user-images.githubusercontent.com/2996042/215875370-7e8f1537-847b-4bfb-92d4-2fe7613a0b67.png)

As alerady said, the PCB tracks for the GND and the mosfet Drain to the led strip should be thicker
I would go for 1.2 to 1.5 mm to cover ~1.5..2A max

How much current will you use ?

**Reply by lionlggnd** [reply]
> Hi,
> 
> since you seem very new in this domain (nothing wrong with that), let me try to avoid some pitfalls I still see. You don't try something exotic, so what you have is likely to work, but not 100%.
> 
> You did not mention which mosfet you will use. There are a _lot_ of them available with different specifications, and you should definitely check their datasheet before ordering/using them (or post the exact type of your candidate here). Look for
> 
> * Pin assignment. 1=Gate, Drain, Source=3 is no standard to really rely on.
> * Switching mosfets with 3.3V as provided by the ESP is not very common: look at R_ds value or diagram in the datasheet. The value itself is not so relevant for your case, but the voltage V_ds given that is needed to get this R_ds. It is very often above the 3.3V but should at least be close. If your R_ds value is 4.5-6V you cannot reach full brightness (but might still be ok). Above that mosfets might not switch on at all at 3.3V.
> * You have the resistors that avoid the leds flashing while the pins of the ESP have no defined voltage during early boot (or you don't want to care). But it might be important to protect the 3 ESP pwm pins as well. Insert resistors between Dx and the junction to your 1k resistors. Why? The gate of the mosfet acts as a small capacitor (again, some mosfets more than others). If you switch the voltage, the capacitor can deliver its charge back to the pins at a high current (although very short), well above what the esp pins are designed for. With ohms law you can check: you want at most ~20mA or so and you have 5V -> R=U/I = 5/0.02 = 250 ohm. The ESP might tolerate it without resistors, but why take chances. I'd go for the common values 220 or 330 ohm.
> * Normal usb ports of pc's are specified to deliver up to 0.5A. You say your strip needs 1A, so take care where you plug it in. Check that your power supply can deliver that. Probably all non pc usb port power sources can do more, but check it anyways.
> 
> Have fun!


![image](https://user-images.githubusercontent.com/117315249/215889016-3fedefe9-1d6e-4c42-9357-0a33f4de3054.png)
the pcb is ok? i can print it?

> Yes, take care of purchaing the TO220 package. It is compatible with your PCB routing the other is SMD
> 
> ![image](https://user-images.githubusercontent.com/2996042/215875370-7e8f1537-847b-4bfb-92d4-2fe7613a0b67.png)
> 
> As alerady said, the PCB tracks for the GND and the mosfet Drain to the led strip should be thicker I would go for 1.2 to 1.5 mm to cover ~1.5..2A max
> 
> How much current will you use ?

Ok, i found [this ](https://it.aliexpress.com/item/4000830214631.html?spm=a2g0o.productlist.main.1.16079de0Vy8Ceu&algo_pvid=5ebb2e4b-794c-4a16-bc37-5dfbd9808d83&algo_exp_id=5ebb2e4b-794c-4a16-bc37-5dfbd9808d83-0&pdp_ext_f=%7B%22sku_id%22%3A%2210000008680933661%22%7D&pdp_npi=2%40dis%21EUR%212.1%212.1%21%21%21%21%21%40211bd93f16751949001517422d0717%2110000008680933661%21sea&curPageLogUid=yLllEBzScrFU)on aliexpress

**Reply by sfromis** [reply]
While the thicker traces for the pullup resistors works just fine, it is not a benefit either, as the current here is very low.

**Reply by joba-1** [reply]
your link goes to the p-channel variant NDP6020P where all polarities are switched. 
You need the NDP6020 (n-channel) that Barbudor suggested. (thanks, I‘ll order both 😀 … or I would if I could find one - n channel on short supply 🤨…)

**Comment by lionlggnd** [comment, marked as answer]
On Aliexpress i can found only the P-channel version. Can i insert the mosfet reverse-side on the pcb o i need to adjust the wiring of the pcb? I can search it on digikey but i prefer avoid to spend 15$ delivery for only 2$ component 🤣🤣🤣

**Reply by lionlggnd** [reply]
if i can fix the routing of the pcb to make it adapt to the p-channel version better.  otherwise let me know some links where I can buy it ... on the product data sheet it says it is an obsolete mosfet.  on aliexpress no trace.

**Reply by barbudor** [reply]
No
If your strip has a common VCC and need to drive the cathode, N channel is the best solution

That's a confusing part reference
I'll have to check I bought the right ones for me and indeed seems difficult to find the N version

**Reply by lionlggnd** [reply]
There is a method to change the routing of the pcb to adapt it to p chan

**Reply by lionlggnd** [reply]
I searched on digikey no results.

**Reply by lionlggnd** [reply]
IRF520 can work?
I saw a video online of a guy used IRF510, and actually worked with the strip.
Let me know.

**Reply by joba-1** [reply]
Same problem. Vgs to open the gate is too high for the esp to deliver. 

regarding "used IRF510, and actually worked": that is possible, or even common, but it depends on luck. Some components already have a good enough Rds at 3.3V although the manufacturer doesn't want to guarantee it. In any case it is an unnecessary compromise: it will have more heat dissipation (wastes energy) and won't be as bright as it could get.

I now did an extensive search on mouser for components that might match (have not checked the details yet, but likely they will all fit your use case).

My filter criteria (so you can see for yourself) was:
All [Products/Semiconductors/Discrete Semiconductors/Transistors/MOSFET](https://www.mouser.de/c/semiconductors/discrete-semiconductors/transistors/mosfet/) (which is over 20000 variants), then filter
* mounting style: through-hole
* transistor polarity: n channel
* continuous drain current > 2A (doubled your requirement or it might get quite hot during normal operation)
* Rds(on) < 50 mOhm
* Vgs(threshold) <= 1V (needs headroom because at that voltage it only barely starts conducting and we want it more or less fully conducting at ~3V)
* in stock (so you can actually get them at least here)

[21 components left](https://mou.sr/40hvqFn)

Datasheets

https://www.mouser.de/datasheet/2/308/1/NDP6060L_D-1812970.pdf
https://www.mouser.de/datasheet/2/427/supsub85-1838723.pdf
https://www.mouser.de/datasheet/2/196/Infineon-IPP042N03L-DS-v02_00-en-522853.pdf
https://www.mouser.de/datasheet/2/308/1/FDPF320N06L_D-1808202.pdf
https://www.mouser.de/datasheet/2/115/DIOD_S_A0006645099_1-2542748.pdf
https://www.mouser.de/datasheet/2/427/91317-1768661.pdf
https://www.mouser.de/datasheet/2/308/1/FDP5800_D-1808068.pdf
https://www.mouser.de/datasheet/2/196/INFNS10791_1-2519731.pdf
https://www.diodes.com/assets/Datasheets/DMT6009LCT.pdf
https://www.mouser.de/datasheet/2/427/91318-1768577.pdf
https://www.mouser.de/datasheet/2/916/PHP191NQ06LT-2938294.pdf
https://www.mouser.de/datasheet/2/196/IPP055N03L_rev1.04-522939.pdf
https://www.mouser.de/datasheet/2/196/IPP055N03L_rev1.04-1732316.pdf
https://www.mouser.de/datasheet/2/196/Infineon-IPP042N03L-DS-v02_00-en-1731857.pdf
https://www.mouser.de/datasheet/2/427/sup40012el-1766368.pdf
https://www.rohm.com/datasheet?p=RX3G07CGN&dist=Mouser&media=referral&source=mouser.com&campaign=Mouser
https://www.mouser.de/datasheet/2/308/1/HUF76423P3_D-1810156.pdf
https://www.diodes.com/assets/Datasheets/DMNH4005SCT.pdf
https://www.mouser.de/datasheet/2/196/infns10791_1-2271144.pdf
https://www.mouser.de/datasheet/2/389/stp60nf06l-1851393.pdf
https://www.diodes.com/assets/Datasheets/DMNH4005SCTQ.pdf

did a quick check (looking at the diagram that tells you which Rds you have for a given Vgs) on the first one and I'd say it fits.
Usually you also check for heat problems: from Rds and Id you can now calculate the power dissipation P = I^2 * R (limit is given in the datasheet but may already include cooling by heatsinks). Since you switch on and off often, your actual Rds can be even higher on average so you need to add a good margin (or calculate it exactly - or depend on luck ;) )

**Reply by lionlggnd** [reply]
so the NDP6060L it fits, is compatible with my pcb and will work?

found on aliexpress.

but for only 1-2A i dont think the mosfet will abnormally heat ...

**Reply by joba-1** [reply]
> but for only 1-2A i dont think the mosfet will abnormally heat

I think you are right. (And I looked it up: dissipation up to 250W and you do below 2W per color - a good margin indeed :) )

The other 20 might be even slightly better or cheaper though but I'm sure this one is fine.

**Reply by lionlggnd** [reply]
ok, i can order it (if unavailable or costly i can order another in the list) and print the pcb?

**Reply by joba-1** [reply]
> ok, i can order it (if unavailable or costly i can order another in the list) and print the pcb?

of course you can order and print. The question is: will you sue me if it doesn't work?

**Reply by lionlggnd** [reply]
no hahahahhahah

**Reply by lionlggnd** [reply]
thanks for your help!

**Reply by joba-1** [reply]
For the record: the FDP5800 are my favorites. Just ordered...

**Reply by lionlggnd** [reply]
Ok, i order too

**Reply by barbudor** [reply]
OMG, I just checked and the 20 NDP6020 I purchased a few monhts ago are actually DNP6020P
So happy that the project I purchased them for is just stalled .... :rofl:

**Reply by lionlggnd** [reply]
🤣🤣🤣🤣

**Reply by joba-1** [reply]
Followup (sorry, kind of late...): 

I finally got around using my FDP5800 in an actual circuit with an ESP32 and found it struggles with 3.3V at the mosfet gate. It works in principle, but LEDs are very dim. Using 5V at the gate and it works fine.

So I switched my favorite now to the IRLB3034. Only 3V at the gate are enough for full brightness.

Tested with 25kHz PWM switching 12V at 1A. Temperature rising barely noticeable. Not surprising since according to datasheet it can do 40V and >10A (with 5V at the gate and cooling even more than 100A)

![grafik](https://github.com/arendst/Tasmota/assets/32450554/8261485e-4468-4db2-ab0e-577cd316ddbb)

This is the circuit I used (actually using more channels for rgbw). 
* R1 is to make sure the LED is off while the ESP is not ready (pin is floating). 
  Connected to the IO pin and not the gate so it doesn't reduce the gate voltage by some millivolts: 
  LED gets brighter, mosfet keeps cooler (a tiny bit).
* R2 is to reduce current from the mosfet gate to the IO pin caused by (very) high pwm frequencies and the gate capacity.
  Probably not needed here and resistance shouldn't be bigger or it slows down switching which makes the mosfet hot.
* R3 is recommended for my type of high power leds. Not needed for strips etc.

**Comment by lionlggnd** [comment]
after configure in tasmota the 3-pwm gpio, the system will auto recognize a led strip?

**Reply by barbudor** [reply]
Read this : https://tasmota.github.io/docs/Lights/

You will get control such as the one presented here : https://tasmota.github.io/docs/Lights/#3-channels-rgb-lights
Control of the colour (hue), saturation and brightness (known as HSB)
You can control from the UI or using commands from an home automation
With hte appropriate Tasmota variant you can also control from you Alexa/GoogleHome...

**Comment by lionlggnd** [comment]
-->thanks

**Comment by lionlggnd** [comment]
Hi!
All components delivered, and effectively works.
But if the strip length increase, the leds gradually become less brighting, and the Green color become gradually totally off.
This is a MOSFET problem (ndp6060 delivered) or i should increase the Amperes of the positive 5v of the strip?
Thanks 😊

**Reply by barbudor** [reply]
How is the led strip and the power supply and the above PCB connected ?
What is the length of the strip ?
Above a certain length the tracks in the strip itself may be not enough to carry the current and it could help to cut the strip in smaller segments that are connected in parallel

**Reply by sfromis** [reply]
You should most likely inject current in more than one spot, as a starting point both ends, but depending on length, maybe more points along the strip. The thing is that 5V, and high brightness on a long strip needs so much current that you cannot expect the traces on the strip to carry it from one end to another, without losing a lot to resistance and heat.

Added: For a plain strip, this is functionally nearly equivalent to what barbudor suggested about cutting up into parallel strips.

**Reply by lionlggnd** [reply]
So, by pulling a cable on the back and bringing it to the +5V input of the continuous strip also on the opposite end, can the strip light up more?

**Reply by sfromis** [reply]
Well, by "power", I meant all pins where power is flowing. That includes the color channels, as these act as ground connections via your MOSFETs. Of course, the 5V may be the most important, as it has to power all the color channels.

**Comment by lionlggnd** [comment]
p.s i'm taking the 5V power from the "VU" pin on the nodemcu that (as the online docs says) can provide 5V.
the nodemcu is directly connected to micro usb port and GPIOs to the PCB
is better if i connect directly the +5V of the strip to the positive of the wall-usb charger?
also because i saw this [thread](https://arduino.stackexchange.com/questions/75645/vin-vu-pins-of-nodemcu-v3).

**Reply by barbudor** [reply]
Try powering the led strip at the opposite of the side connected to the mosfet
With a proper cable awg 20 or bigger

**Reply by sfromis** [reply]
If you want to draw more than 1A, you should not get it from a NodeMCU pin.

**Reply by lionlggnd** [reply]
Thanks! i'll try. :)
So is this a powering problem, not the nodemcu/ ndp6060 or oth.
(This problem happen only when increasing strip length).
after doing this, should work?

**Reply by sfromis** [reply]
Well, odds improve if you get to fix weaknesses in your handing of current flow, but just one weak link may be where it stops. Anywhere more than minor current flows, you should consider how much current you need, and how much current the connection can handle. This is especially critical at a low voltage like 5V, as a wire able to carry 0.5A can provide 2.5W with 5V, but 12W with 24V

**Comment by lionlggnd** [comment]
Hi, i have a 5V strip, but with the ndp, it is very very dim.

Switching with IRLB3034 might do the trick?
Thanks!!!

**Reply by joba-1** [reply]
Yes. 
The voltage to switch between drain and source is irrelevant for this, your 5V or my 12V. So it seems you have exactly the same problem I had: 3.3V at the gate is not enough to switch the mosfet fully on.

**Reply by joba-1** [reply]
Unfortunately I can share a bad experience: I recommended the irlb3034 based on datasheet and a test setup. This still holds, But after that I ordered more and got a bad batch: only 2 of 20 mosfets opened at 3.3V, all others only at around 5V. It was cheap and I got a full refund, but anyways: not a nice experience. Be prepared... ;)

**Reply by lionlggnd** [reply]
Ok, but if the datasheet says it, the power mosftet MUST open at 3,3V. Otherwise, the mosfet is defective. Where you bought the mosfet?

**Reply by lionlggnd** [reply]
If i order that mosfet, how i can test it to see if opens at 3,3?

**Reply by joba-1** [reply]
yes, it is faulty. If you used a local dealer or known electronics provider like Mouser or Reichelt, that is unlikely to happen. I ordered from an unknown (to me) seller on Aliexpress.

**Reply by joba-1** [reply]
How to test depends on what equipment you have. A multimeter to measure voltage or amperes and a potentiometer would be good. Some resistors and a led might also work as a simple ok test.

**Reply by joba-1** [reply]
This simple test (similar to infineon datasheet fig23a on page 7) worked for me:
Given you have a multimeter and a 1k resistor, connect the resistor like this with the mosfet and the 5V and 3.3V from your NodeMCU:

5V ---|1k|---D S---Gnd

Now measure voltage between D and S while 

1) Connect Gnd to G: should measure 5V -> mosfet is off
2) Connect 5V to G: should measure 0V -> mosfet is on
   Now you know your circuit is ok and switches the mosfet fully on and off at least with 5V at the gate
3) Connect 3.3V to G: should still measure 0V -> OK

Why? 
The resistance between D and S should be roughly 1 mOhm when on, but I'd also accept 10 mOhm as ok. So about 1/100000 of the 1k resistor.
So the resistor gets the same relative amount of voltage: 5V * 1/100000 = 50µV (less than my average multimeter can measure -> shows 0V)
1 Ohm (1/1000 of 1k) between D and S with 3.3V at G would already be bad and would show: 5V * 1/1000 = 5mV. My multimeter showed something similar and more for the bad mosfets.

If you had a potentiometer, you could (instead of the connection to G above) connect one end to 5V, the other to Gnd and the variable middle pin to G.
Begin with the variable end turned to Gnd (multimeter shows 5V between D and S) and turn it slowly to 5V. Stop when the multimeter shows 0V. Now measure the voltage between Gnd and the middle potentiometer pin -> The voltage where the gate switches the mosfet fully on. Should be less than 3V.

**Reply by lionlggnd** [reply]
I can buy only on aliexpress, bc shipping fee in others stores online (here in italy the most popular is tme) is very very high and expensive, like 10 times the cost of the original product.
Please link me the defective article so i don’t buy that article 😊

**Reply by joba-1** [reply]
same here :) 

store is Si Tai
https://a.aliexpress.com/_EQNJhGf

product was this faulty item on AliExpress.  1,64€  10％ Off | 10PCS IRLB3034 IRLB3036 IRLB3813 IRLB4030 IRLB4132 IRLB8314 IRLB8721 IRLB8743 TO220 TO-220
https://a.aliexpress.com/_EQL91F9

**Reply by lionlggnd** [reply]
Bought on Aliexpress (TME has 7€ of shipping fee…) from another store, lets hope it works. I’ll lyk

**Comment by lionlggnd** [comment]
Thanks!!Will buy and let you know.
